addOns/graphql/src/main/java/org/zaproxy/addon/graphql/GraphQlParser.java

```diff
@@ -19,49 +19,108 @@
-import org.parosproxy.paros.control.Control;
-import org.parosproxy.paros.extension.history.ExtensionHistory;
-import org.parosproxy.paros.model.HistoryReference;
-import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.Constant;
 import org.parosproxy.paros.network.HttpMessage;
-import org.parosproxy.paros.network.HttpSender;
-import org.zaproxy.zap.utils.ThreadUtils;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.zaproxy.zap.network.HttpRequestBody;
 
 public class GraphQlParser {
 
     private static final Logger LOG = Logger.getLogger(GraphQlParser.class);
+    private final URI endpointUrl;
+    private final int initiator;
+    private final RequestHandler requestHandler;
+    private HttpMessage importMessage;
+
+    public GraphQlParser(String endpointUrlStr, int initiator) throws URIException {
+        this(UrlBuilder.build(endpointUrlStr), initiator);
+    }
+
+    public GraphQlParser(URI endpointUrl, int initiator) {
+        this.endpointUrl = endpointUrl;
+        this.initiator = initiator;
+        requestHandler = new RequestHandler(initiator);
+    }
+
+    public void introspect() throws IOException {
+        JsonObject msgBodyJson = new JsonObject();
+        msgBodyJson.addProperty("query", IntrospectionQuery.INTROSPECTION_QUERY);
+        HttpRequestBody msgBody = new HttpRequestBody(msgBodyJson.toString());
+
+        HttpRequestHeader msgHeader =
+                new HttpRequestHeader(HttpRequestHeader.POST, endpointUrl, "HTTP/1.1");
+        msgHeader.setHeader("Accept", "application/json");
+        msgHeader.setHeader("Content-Type", "application/json");
+        msgHeader.setContentLength(msgBody.length());
+
+        importMessage = new HttpMessage(msgHeader, msgBody);
+        requestHandler.makeRequest(importMessage);
 
-    public static void parse(URI uri) {
-        HttpMessage msg;
         try {
-            msg = new HttpMessage(uri);
-            HttpSender sender =
-                    new HttpSender(
-                            Model.getSingleton().getOptionsParam().getConnectionParam(),
-                            true,
-                            HttpSender.MANUAL_REQUEST_INITIATOR);
-            sender.sendAndReceive(msg, true);
-        } catch (Exception e) {
-            LOG.error("Unable to send request.", e);
-            return;
+            Map<String, Object> result =
+                    new Gson()
+                            .fromJson(
+                                    importMessage.getResponseBody().toString(),
+                                    new TypeToken<Map<String, Object>>() {}.getType());
+            @SuppressWarnings("unchecked")
+            Document schema =
+                    new IntrospectionResultToSchema()
+                            .createSchemaDefinition((Map<String, Object>) result.get("data"));
+            String schemaSDL = new SchemaPrinter().print(schema);
+            Thread t = new Thread(() -> parse(schemaSDL));
```

Comment:
- Name the thread with `ZAP-` prefix to be easier to know that's a ZAP thread (e.g. `ZAP-GraphQL-Parser`).
 - The thread was not started in the `importUrl`, better extract a method that does this to avoid duplicating in each of them.